docs/language-server/protocol-language-server.md

```diff
@@ -5333,6 +5333,43 @@ null;
 - [`FileSystemError`](#filesystemerror) to signal a generic, unrecoverable
   file-system error.
 
+## Runtime Operations
+
+### `runtime/getComponentGroups`
+
+Sent from the client to the server to get the list of component groups available
+in runtime.
+
+The engine is started with an empty list of libraries loaded. It means that the
+request should be sent after the first
+[`executionContext/executionComplete`](#executioncontextexecutioncomplete)
+notification indicating that all the libraries are loaded, and the component
+group list is populated. If the request is sent before the first notification,
+the response may be empty or not contain all available components.
```

Comment:
Should we have an additional field indicating that the list is complete (or not)?

Reply:
I'm not sure if we can do that. Right now IDE relies on the `executionComplete` notification to make sure that all the libraries are loaded.